app/src/main/proto/vss.proto

```diff
@@ -73,18 +73,26 @@ message PutObjectRequest {
   // a database-transaction in an all-or-nothing fashion.
   // All Items in a single `PutObjectRequest` must have distinct keys.
   //
-  // Clients are expected to store a `version` against every `key`.
-  // The write will succeed if the current DB version against the `key` is the same as in the request.
-  // When initiating a `PutObjectRequest`, the request should contain their client-side version for
-  // that key-value.
-  //
-  // For the first write of any key, the `version` should be '0'. If the write succeeds, the client
-  // must increment their corresponding key versions (client-side) by 1.
-  // The server increments key versions (server-side) for every successful write, hence this
-  // client-side increment is required to ensure matching versions. These updated key versions should
-  // be used in subsequent `PutObjectRequest`s for the keys.
+  // Key-level versioning (Conditional Write):
+  //   Clients are expected to store a `version` against every `key`.
+  //   The write will succeed if the current DB version against the `key` is the same as in the request.
+  //   When initiating a `PutObjectRequest`, the request should contain their client-side `version`
+  //   for that key-value.
   //
-  // Requests with a conflicting version will fail with `CONFLICT_EXCEPTION` as ErrorCode.
+  //   For the first write of any `key`, the `version` should be '0'. If the write succeeds, the client
+  //   must increment their corresponding key versions (client-side) by 1.
+  //   The server increments key versions (server-side) for every successful write, hence this
+  //   client-side increment is required to ensure matching versions. These updated key versions should
+  //   be used in subsequent `PutObjectRequest`s for the keys.
+  //
+  //   Requests with a conflicting/mismatched version will fail with `CONFLICT_EXCEPTION` as ErrorCode
+  //   for conditional writes.
+  //
+  // Skipping key-level versioning (Non-conditional Write):
+  //   If you wish to skip key-level version checks, set the `version` against the `key` to '-1'.
+  //   This will perform a non-conditional upsert query. After any non-conditional upsert, the
+  //   `version` against the `key` is reset to '1'. Hence the next `PutObjectRequest` for the `key`
```

Comment:
nit: Merge these two sentences as:

> This will perform a non-conditional upsert query, after which the `version` against the `key` is reset to '1'.